src/Symfony/Component/Validator/Constraints/NegativeValidator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * Validates that values are negative.
+ *
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
+ */
+class NegativeValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Negative) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Negative');
+        }
+
+        if (!(\is_int($value) || \is_float($value) || \is_string($value))) {
```

Comment:
i would invert it `!is_int() && !is_float() && !is_string()`

Reply:
Much more readable 👍